google/cloud/aiplatform/training_jobs.py

```diff
@@ -15,9 +15,309 @@
+
+        # The root module of the python package
+        trainer_path = os.path.join(trainer_root_path, self._ROOT_MODULE)
+
+        # __init__.py path in root module
+        init_path = os.path.join(trainer_path, "__init__.py")
+
+        # The module that will contain the script
+        script_out_path = os.path.join(trainer_path, f"{self._TASK_MODULE_NAME}.py")
+
+        # The path to setup.py in the package.
+        setup_py_path = os.path.join(trainer_root_path, "setup.py")
+
+        # The path to the generated source distribution.
+        source_distribution_path = os.path.join(
+            trainer_root_path,
+            "dist",
+            f"{self._ROOT_MODULE}-{self._SETUP_PY_VERSION}.tar.gz",
+        )
+
+        # Make required dirs
+        os.mkdir(trainer_root_path)
+        os.mkdir(trainer_path)
+
+        # Make empty __init__.py
+        with open(init_path, "w") as fp:
+            pass
+
+        # Format the setup.py file.
+        setup_py_output = self._SETUP_PY_TEMPLATE.format(
+            name=self._ROOT_MODULE,
+            requirements=",".join(f'"{r}"' for r in self.requirements),
+            version=self._SETUP_PY_VERSION,
+        )
+
+        # Write setup.py
+        with open(setup_py_path, "w") as fp:
+            fp.write(setup_py_output)
+
+        # Copy script as module of python package.
+        shutil.copy(self.script_path, script_out_path)
+
+        # Run setup.py to create the source distribution.
+        setup_cmd = self._SETUP_PY_SOURCE_DISTRIBUTION_CMD.format(
+            python3=_get_python3_alias()
+        ).split()
+
+        p = subprocess.Popen(
+            setup_cmd,
+            cwd=trainer_root_path,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        output, error = p.communicate()
+
+        # Raise informative error if packaging fails.
+        if p.returncode != 0:
+            raise RuntimeError(
+                "Packaging of training script failed with code %d\n%s \n%s"
+                % (p.returncode, output.decode(), error.decode())
+            )
+
+        return source_distribution_path
+
+    def package_and_copy(self, copy_method: Callable[[str], str]) -> str:
+        """Packages the script and executes copy with given copy_method.
+
+        Args:
+            copy_method Callable[[str], str]
+                Takes a string path, copies to a desired location, and returns the
+                output path location.
+        Returns:
+            output_path str: Location of copied package.
+        """
+
+        script_name = os.path.basename(self.script_path)
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            source_distribution_path = self.make_package(tmpdirname)
+            return copy_method(source_distribution_path)
+
+    def package_and_copy_to_gcs(
+        self,
+        staging_bucket: str,
+        project: str = None,
```

Comment:
```suggestion
        project_id: str = None,
```
If this accepts both project numbers and ids please disregard.

Reply:
Ack.